salt/modules/ssh.py

```diff
@@ -149,15 +149,8 @@ def _replace_auth_key(
                     # Commented Line
                     lines.append(line)
                     continue
-                comps = line.split()
-                if len(comps) < 2:
-                    # Not a valid line
-                    lines.append(line)
-                    continue
-                key_ind = 1
-                if comps[0][:4:] not in ['ssh-', 'ecds']:
-                    key_ind = 2
-                if comps[key_ind] == key:
+                comps = re.findall(r'((.*)\s)?(ssh-[a-z0-9-]+|ecdsa-[a-z0-9-]+)\s([a-zA-Z0-9+/]+={0,2})(\s(.*))?', line)
```

Comment:
@cachedout to be honest, I don't feel qualified for that. I'm not much of a developer and as this function is in the business of reading and writing a file, I'd probably just mess things up. The same goes for refactoring the matching and reading/writing.
Though if you could give me a pointer where to steal the test from, I could give it a shot. 
Or I can come up with some test data if you (or someone else) would be willing to integrate it.